repo_policy_compliance/check.py

```diff
@@ -65,6 +69,41 @@ class Report(NamedTuple):
 
 log.setup()
 
+P = ParamSpec("P")
+R = TypeVar("R")
+
+
+def github_exceptions_to_fail_report(func: Callable[P, R]) -> Callable[P, R | Report]:
+    """Catch exceptions and convert to failed report with reason.
+
+    Args:
+        func: The function to catch the GithubClient exceptions for.
+
+    Returns:
+        The function where any exceptions raised would be converted to a failed result.
+    """
+
+    @functools.wraps(func)
+    def wrapper(*args: P.args, **kwargs: P.kwargs) -> R | Report:
+        """Replace function.
+
+        Args:
+            args: The positional arguments passed to the original method.
+            kwargs: The keywords arguments passed to the original method.
+
+        Returns:
+            Failed result report if any exceptions were raised. The return value after calling the
+            wrapped function otherwise.
+        """
+        try:
+            return func(*args, **kwargs)
+        except GithubClientError as exc:
+            return Report(result=Result.ERROR, reason=str(exc))
+        except ConfigurationError as exc:
+            return Report(result=Result.ERROR, reason=str(exc))
```

Comment:
As mentioned in another comment, I would not set the reason dynamically from the exception.
We have no control over the content (which may be sensitive) and in general internal server error messages should not be detailed for end users. They should include a message to contact the operator (who should have logs) to fix the problem.

Reply:
Fixed with a static error message to contact the operator, thanks!

Reply:
Actually, it turns out that it's okay to `str(exc)` because in `github_client.py`, the exception is raised with messages that can be propagated to the user. Other errors are hidden behind "The github client encountered an error.". I've added a logging for other errors.